x-pack/plugins/lens/public/xy_visualization/xy_config_panel/threshold_panel.tsx

```diff
@@ -0,0 +1,289 @@
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        display="columnCompressed"
+        fullWidth
+        label={i18n.translate('xpack.lens.xyChart.lineThickness.label', {
+          defaultMessage: 'Line thickness',
+        })}
+      >
+        <LineThicknessSlider
+          value={currentYConfig?.lineWidth || 1}
+          onChange={(value) => {
+            // TODO: fix number validation
+            setYConfig({ lineWidth: value });
+          }}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        display="columnCompressed"
+        fullWidth
+        label={i18n.translate('xpack.lens.xyChart.fillThreshold.label', {
+          defaultMessage: 'Fill',
+        })}
+      >
+        <EuiButtonGroup
+          isFullWidth
+          legend={i18n.translate('xpack.lens.xyChart.fillThreshold.label', {
+            defaultMessage: 'Fill',
+          })}
+          data-test-subj="lnsXY_fill_threshold"
+          name="fill"
+          buttonSize="compressed"
+          options={[
+            {
+              id: `${idPrefix}none`,
+              label: i18n.translate('xpack.lens.xyChart.fillThreshold.none', {
+                defaultMessage: 'None',
+              }),
+              'data-test-subj': 'lnsXY_fill_none',
+            },
+            {
+              id: `${idPrefix}above`,
+              label: i18n.translate('xpack.lens.xyChart.fillThreshold.above', {
+                defaultMessage: 'Above',
+              }),
+              'data-test-subj': 'lnsXY_fill_above',
+            },
+            {
+              id: `${idPrefix}below`,
+              label: i18n.translate('xpack.lens.xyChart.fillThreshold.below', {
+                defaultMessage: 'Below',
+              }),
+              'data-test-subj': 'lnsXY_fill_below',
+            },
+          ]}
+          idSelected={`${idPrefix}${currentYConfig?.fill || 'none'}`}
+          onChange={(id) => {
+            const newMode = id.replace(idPrefix, '') as FillStyle;
+            setYConfig({ fill: newMode });
+          }}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        display="columnCompressed"
+        fullWidth
+        label={i18n.translate('xpack.lens.xyChart.axisSide.icon', {
+          defaultMessage: 'Icon',
+        })}
+      >
+        <IconSelect
+          value={currentYConfig?.icon}
+          onChange={(newIcon) => {
+            setYConfig({ icon: newIcon });
+          }}
+        />
+      </EuiFormRow>
+    </>
+  );
+};
+
+const LineThicknessSlider = ({
+  value,
+  onChange,
+}: {
+  value: number;
+  onChange: (value: number) => void;
+}) => {
+  const { inputValue, handleInputChange } = useDebouncedValue({ value, onChange });
+  return (
+    <EuiRange
+      fullWidth
+      data-test-subj="lnsXY_lineThickness"
+      value={inputValue}
+      showInput
+      min={1}
+      max={10}
+      step={1}
+      compressed
+      onChange={(e) => {
+        handleInputChange(Number(e.currentTarget.value));
+      }}
+    />
```

Comment:
If inputing a line thickness lower than the minimum (ex. `0`) or greater than the maximum (ex. `100`) supported value, the input is shown in an invalid state. Instead, can we just automatically change the number on blur to the appropriate minimum or maximum supported value automatically? This saves the user the trouble of doing it themselves, while also educating them on the limits.